org.eclipse.dawnsci.nexus/src/org/eclipse/dawnsci/nexus/NexusUtils.java

```diff
@@ -543,4 +547,176 @@ private static void recursivelyLoadNexusTree(NexusFile nexusFile, GroupNode grou
+
+					String axis = null;
+					int pos = -1;
+					boolean isSignal = false;
+					while (att.hasNext()) {
+						Attribute attribute = att.next();
+						if ("axis".equals(attribute.getName())) {
+							int iaxis = getAttributeIntValue(attribute);
+							if (iaxis < 0) { // We look for comma separated string
+								final int[] axesDims = attribute.getShape();
+
+								final int[] shapeAxes = dataNode.getDataset().getShape();
+								final int[] shapeData = shape;
+								if (axesDims != null && axesCompatible(axesDims, shapeAxes, shapeData)) {
+									for (int dim : axesDims) {
+										if (dim == dimension) {
+											axis = parentPath + Node.SEPARATOR + dataNode.getDataset().getName() + ":" + dimension;
+											break;
+										}
+									}
+								}
+
+							}
+							if (iaxis == dimension) {
+
+								final int[] dims = dataNode.getDataset().getShape();
+
+								if ((dims.length == 1 && dims[0] == size) || dims.length != 1) {
+									axis = parentPath + Node.SEPARATOR + dataNode.getDataset().getName();
+								}
+
+							}
+
+						} else if ("primary".equals(attribute.getName())) {
+							if (pos != 0)
+								pos = getAttributeIntValue(attribute);
+
+						} else if ("label".equals(attribute.getName())) {
+							int labelAxis = getAttributeIntValue(attribute);
+							if (labelAxis == dimension)
+								pos = 0;
+
+						} else if ("signal".equals(attribute.getName())) {
+							isSignal = true;
+							axis = null;
+							pos = -1;
+							break;
+						}
+					}
+
+					// prioritise datasets that specify an axes (even with no primary attribute)
+					// over other datasets
+					if (axis != null && pos == -1) {
+						pos = fakePosValue--;
+					}
+
+					// Add any the same shape as this dimension
+					// providing that they are not signals
+					// Some nexus files set axis wrong
+					if (axis == null && !isSignal) {
+						final int[] dims = dataNode.getDataset().getShape();
+						if (dims[0] == size && dims.length == 1) {
+							axis = parentPath + Node.SEPARATOR + dataNode.getDataset().getName();
+						}
+					}
+
+					if (axis != null) {
+						if (pos < 0) {
+							axesTmp.add(axis);
+						} else {
+							axesMap.put(pos, axis);
+						}
+					}
+				}
+			}
+
+			final List<String> axes = new ArrayList<String>(3);
+			if (!axesMap.isEmpty()) {
+				for (Integer pos : axesMap.keySet()) {
+					axes.add(axesMap.get(pos));
+				}
+			}
+			axes.addAll(axesTmp);
+
+			return axes;
+		}
+	}
+
+	/**
+	 * Gets the int value or returns -1 (Can only be used for values which are not
+	 * allowed to be -1!)
+	 * 
+	 * @param attribute
+	 * @return
+	 */
+	private static int getAttributeIntValue(Attribute attribute) {
+		final Object ob = attribute.getValue();
+		if (ob instanceof int[]) {
+			int[] ia = (int[]) ob;
+			return ia[0];
+		} else if (ob instanceof String[]) {
+			String[] sa = (String[]) ob;
+			try {
+				return Integer.parseInt(sa[0]);
+			} catch (Throwable ne) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)